keps/6757-failure-recovery/README.md

```diff
@@ -0,0 +1,336 @@
+* This feature introduces a potential footgun to Kueue users and could turn out to be to volatile/risky to use in most affected cases.
+* It introduces yet another responsibility for Kueue - on top of quota management and scheduling, it will also start performing failure recovery.
+
+## Alternatives
+
+### Do Nothing
+
+Since this feature spawned a lengthy discussion about its riskiness and there are other controllers in the ecosystem which deal with
+node problem remediation (e.g. [medik8s](https://github.com/medik8s/self-node-remediation)), an alternative would be to do nothing,
+wait for the upstream conversations to be resolved and propose an alternative solution (external to Kueue) to the affected users.
+
+#### Advantages
+
+1. Requires no implementation effort.
+
+#### Drawbacks
+
+1. Forces users to run another system alongside Kueue, making deploying Kueue more complex those cases.
+
+### Hide The Controller Behind A Feature Gate
+
+Instead of introducing changes to the API, a feature gate could be used to get some initial feedback about the
+feature and its risks.
+
+#### Advantages
+
+1. Easy to implement, less moving parts.
+1. Allows to gather feedback quickly.
+
+#### Drawbacks
+
+1. It won't allow to implement the pod selectors and configurations for the failure recovery,
+limiting the implementation to defaults:
+  1. `podLabelSelector` - everything.
+  1. `forcefulTerminationGracePeriod` - a constant or a multiple of `terminationGracePeriodSeconds`.
+
+### Managing The `node.kubernetes.io/out-of-service` Taint On `Node`
+
+Core Kubernetes already contains logic for [automatic garbage collection](https://github.com/kubernetes/kubernetes/blob/4870d987d0a4aac2d9223d4c0b9f22858c0d1590/pkg/controller/podgc/gc_controller.go) of `Pod`s running on a `Node` with the `node.kubernetes.io/out-of-service` taint.
+It updates the `Pod`s status and deletes it from `etcd`. The recovery controller could use this fact to terminate stuck pods by automatically adding this taint to nodes which are not ready for some configurable time.
+
+Given the API proposal, this approach could potentially be implemented as an alternative
+ recovery strategy in the future.
+
+ ```go
+type OutOfServiceNodeTaintingConfig struct {
+  NotReadyNodeTimeout *time.Duration `json:"notReadyNodeTimeout"`
+}
+
+type FailureRecoveryAction string
+
+const (
+  PodTermination FailureRecoveryAction = "PodTermination"
+  TaintingNodeOutOfService FailureRecoveryAction = "TaintingNodeOutOfService"
+)
+ ```
+
+#### Advantages
+
+1. It uses an existing Kubernetes mechanism.
+2. Concerns are tidily separated - the recovery controller simply says the node is out of service, the garbage collector in the control plane can decide how and when to clean it up.
+
+
+#### Drawbacks
```

Comment:
nit: I would suggest not to use the "Drawbacks" subsection, because there is already such section referring to the selected approach. Just to avoid a reader getting tricked into counting the nesting levels. For the "Alterneatives" section we usually just have the "reasons to defer" paragraph (take a look at the TAS KEP).